test/extended/testdata/gssapi/fedora/base/gssapi-tests.sh

```diff
@@ -0,0 +1,280 @@
+        os::cmd::expect_failure_and_not_text 'oc whoami' $u
+
+        os::cmd::expect_failure_and_text "oc login -u $u -p wrongpassword" 'Login failed (401 Unauthorized)'
+        os::cmd::expect_failure_and_not_text 'oc whoami' $u
+
+        os::cmd::expect_success_and_text "oc login -u $full -p password" 'Login successful.'
+        os::cmd::expect_success_and_text 'oc whoami' $u
+        os::cmd::expect_success_and_text 'oc logout' $u
+
+        os::cmd::expect_success_and_text "oc login -u $u -p password" 'Login failed (401 Unauthorized)'
+        os::cmd::expect_failure_and_not_text 'oc whoami' $u
+    done
+fi
+
+# Client has GSSAPI and server is GSSAPI only
+# Only GSSAPI works
+
+if [[ -n CLIENT_HAS_GSSAPI && -z SERVER_HAS_BASIC ]]; then
+    for u in "${users[@]}"; do
+        os::cmd::expect_failure "kinit $u <<< wrongpassword"
+        os::cmd::expect_failure_and_text 'oc login' 'No Kerberos credentials available'
+        os::cmd::expect_failure_and_not_text 'oc whoami' $u
+        os::cmd::expect_success 'kdestroy'
+
+        os::cmd::expect_success "kinit $u <<< password"
+        os::cmd::expect_success_and_text 'oc login' 'Login successful.'
+        os::cmd::expect_success_and_text 'oc whoami' $u
+        os::cmd::expect_success_and_text 'oc logout' $u
+        os::cmd::expect_success 'kdestroy'
+
+        os::cmd::expect_failure_and_text "oc login -u $u -p wrongpassword" "Can't find client principal $u in cache collection"
+        os::cmd::expect_failure_and_not_text 'oc whoami' $u
+
+        os::cmd::expect_failure_and_text "oc login -u $u -p password" "Can't find client principal $u in cache collection"
+        os::cmd::expect_failure_and_not_text 'oc whoami' $u
+
+        # Password is ignored if you have the ticket for the user
+        os::cmd::expect_success "kinit $u <<< password"
+        os::cmd::expect_success_and_text "oc login -u $u -p wrongpassword" 'Login successful.'
+        os::cmd::expect_success_and_text 'oc whoami' $u
+        os::cmd::expect_success_and_text 'oc logout' $u
+        os::cmd::expect_success 'kdestroy'
+    done
+
+    # Having multiple tickets
+    os::cmd::expect_success 'kinit user1 <<< password'
+    os::cmd::expect_success 'kinit user2 <<< password'
+    os::cmd::expect_success 'kinit user3 <<< password'
+
+    os::cmd::expect_success_and_text 'oc login -u user1' 'Login successful.'
+    os::cmd::expect_success_and_text 'oc whoami' 'user1'
+    os::cmd::expect_success_and_text 'oc login -u user2' 'Login successful.'
+    os::cmd::expect_success_and_text 'oc whoami' 'user2'
+    os::cmd::expect_success_and_text 'oc login -u user3' 'Login successful.'
+    os::cmd::expect_success_and_text 'oc whoami' 'user3'
+
+    # Ignore password
+    os::cmd::expect_success_and_text 'oc login -u user1 -p wrongpassword' 'Login successful.'
+    os::cmd::expect_success_and_text 'oc whoami' 'user1'
```

Comment:
always use the full usernames in whoami checks to make sure we get realm-qualified even if we don't specify it in -u